### ISSUE TITLE
MySQL-Socket-Issue on CentOS6

### ISSUE THREAD

**Issue by seteq** [inverse-inc/packetfence#1196]
I tried to install Packetfence on a clean and updated CentOS6 by using yum.
I enabled VLAN enforcement and added the three required network interfaces.
On the next page the install wizard notified me that mysql has not been started.
After starting mysql I tried to connect as root without password which raised an error.
On the command line connecting mysql as root just worked fine.

I started digging in **/usr/local/pf/logs/httpd.admin.log** and here it says:
`ERROR: Caught error DBI connect('database=pf;host=localhost;port=3306','pf',...) failed: Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (13)`

So it's trying to use the socket. That would be fine, but unfortunately CentOS seems to be a bit restrictive on that socket file in /var/lib/mysql

```
drwxr-xr-x+  5 mysql mysql     4096 Feb 15 06:59 .
drwxr-xr-x+ 24 root  root      4096 Feb 15 05:59 ..
srwxr-xr-x   1 mysql mysql        0 Feb 15 06:49 mysql.sock
```

So I changed permissions to 0777 on that socket file and then I was able to proceed the install wizard.

It seems a bit strange for CentOS to have a limitation so that only the mysql user can use the socket. I wasn't able to find a way of changing the default permissions of the mysql socket. Whenever you restart the mysql server it gets recreated with 0755 and blocking access.
As an addition the mysql client library internally re-routes connections to the hostname "localhost" directly to the socket, which does not work in that case...

So I ended in changing the mysql hostname to 127.0.0.1 in pfconfig.conf
Please consider changing mysql 'localhost' to '127.0.0.1'

Thanks

**Comment by julsemaan** [comment]
The solution won't be to change it to 127.0.0.1 as it previously worked with the socket and so we must keep it this way.

Please post the output of the following command : 
`# sestatus`

**Comment by seteq** [comment]
SELinux status:                 disabled

Take a look at the permissions on that mysql socket file. The httpd-user "pf" will never be able to write data to that socket. Unless you know how to force mysql to make the socket globally writeable I see no other way than changing that. Any ideas?

**Comment by julsemaan** [comment]
On all the machines I checked, it is 777

Try removing the socket, then restarting the mysql process, then check the permissions

Also please post the content of `/etc/my.cnf`

Also the output of :
`# rpm -qa 'mysql-*'`

**Comment by seteq** [comment]
The CentOS-Machine is running as LXC container - does that make any difference?

When I stop mysqld the socket gets removed anyway. When I delete that socket and restart mysqld it's recreated with 0755.

I did not change anything in the my.cnf file

```
[mysqld]
datadir=/var/lib/mysql
#socket=/var/lib/mysql/mysql.sock
user=mysql
# Disabling symbolic-links is recommended to prevent assorted security risks
symbolic-links=0

[mysqld_safe]
log-error=/var/log/mysqld.log
pid-file=/var/run/mysqld/mysqld.pid
```

```
# rpm -qa 'mysql-*'
mysql-server-5.1.73-5.el6_6.x86_64
mysql-libs-5.1.73-5.el6_6.x86_64
```

EDIT: I should add that I'm also running Ubuntu-servers on the same LXC-Node and their mysql.sock gets created with 0777

**Comment by julsemaan** [comment]
It could be the fact that its in a LXC.

What is the host OS. I'm assuming its also CentOS 6 ?

**Comment by seteq** [comment]
The host OS is Debian 8.3 and I just noticed now that it seems to have AppArmor enabled.
Let me just check if that's the issue, I'll post an update as soon as I know what's going on here...

**Comment by julsemaan** [comment]
I can almost guarantee you this is your problem.

IMO, you shouldn't install a different distribution on your host since they will share the same kernel.

Its already quite a job making all the different services work together on a bare CentOS, don't shoot yourself in the foot by hosting CentOS on your Debian LXC host :)

**Comment by seteq** [comment]
It's almost confirmed that it's a AppArmor issue on the host. I found other users which seem to have similar issues with readonly unix sockets and LXC - see [here](https://forum.proxmox.com/threads/permission-error-w-sockets-inside-ct-since-migration-to-pve-4-1.25244/).

I just did a quick test-drive of packetfence on our LXC node and wondered about the problems which forced me to further investigate that. Now I'll do the install again and use a hardware virtualized VM.

Of course you're right, I don't want to shoot myself in the foot but now it's documented here for anyone running into the same issue ;)

Anyway, thank you for your fast response and your help!
